### PR TITLE
Created thermostatCluster

### PR DESCRIPTION
**Pull request by mhaid**
Adds support for the HAVAC Thermostat Cluster (0x0201)

### CODE REVIEW

File: lib/clusters/thermostat.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const Cluster = require('../Cluster');
+const { ZCLDataTypes } = require('../zclTypes');
+
+const ATTRIBUTES = {
+  localTemperature: { id: 0, type: ZCLDataTypes.int16 },
+  outdoorTemperature: { id: 1, type: ZCLDataTypes.int16 },
+  ocupancy: { id: 2, type: ZCLDataTypes.map8 },
```
Comment (RobinBol) [inline lib/clusters/thermostat.js:9]: You need to provide the right values for the `map8` data type, check groups attribute `nameSupport` for example.

File: lib/clusters/thermostat.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  occupiedHeatingSetpoint: { id: 18, type: ZCLDataTypes.int16 },
+  unoccupiedCoolingSetpoint: { id: 19, type: ZCLDataTypes.int16 },
+  unoccupiedHeatingSetpoint: { id: 20, type: ZCLDataTypes.int16 },
+  minHeatSetpointLimit: { id: 21, type: ZCLDataTypes.int16 },
+  maxHeatSetpointLimit: { id: 22, type: ZCLDataTypes.int16 },
+  minCoolSetpointLimit: { id: 23, type: ZCLDataTypes.int16 },
+  maxCoolSetpointLimit: { id: 24, type: ZCLDataTypes.int16 },
+  minSetpointDeadBand: { id: 25, type: ZCLDataTypes.int8 },
+  remoteSensing: { id: 26, type: ZCLDataTypes.map8 },
+  controlSequenceOfOperation: { id: 27, type: ZCLDataTypes.enum8 },
```
Comment (RobinBol) [inline lib/clusters/thermostat.js:28]: You need to provide the right values for the `enum8` data type, check other clusters for how this works.

File: lib/clusters/thermostat.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  systemMode: { id: 28, type: ZCLDataTypes.enum8 },
```
Comment (RobinBol) [inline lib/clusters/thermostat.js:29]: You need to provide the right values for the `enum8` data type, check other clusters for how this works.

File: lib/clusters/thermostat.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  alarmMask: { id: 29, type: ZCLDataTypes.map8 },
```
Comment (RobinBol) [inline lib/clusters/thermostat.js:30]: You need to provide the right values for the map8 data type, check groups attribute nameSupport for example.

File: lib/clusters/thermostat.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+const COMMANDS = {
+  setSetpoint: {
+    id: 0,
+    args: {
+      mode: ZCLDataTypes.enum8,
```
Comment (RobinBol) [inline lib/clusters/thermostat.js:37]: You need to provide the right values for the enum8 data type, check other clusters for how this works.

**Comment by RobinBol** [comment]
@mhaid just a heads up, in the meantime master has been updated which resulted in a few conflicts in your PR.

**Comment by RobinBol** [comment]
@mhaid as you can see I went ahead and fixed the open issues with your PR so we could merge it, thanks for your contribution!

**Comment by mhaid** [comment]
@RobinBol thank you for the fix and the merge. I will consider the different _enum_ and _map_ usage next time.

Cheers
Morris

**Pull request merged**